pkg/controller/multiarchbuildconfig/util.go

```diff
@@ -0,0 +1,28 @@
+package multiarchbuildconfig
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "github.com/openshift/ci-tools/pkg/api/multiarchbuildconfig/v1"
+)
+
+func getCondition(mabc *v1.MultiArchBuildConfig, condType string) *metav1.Condition {
+	for i := range mabc.Status.Conditions {
+		c := &mabc.Status.Conditions[i]
+		if c.Type == condType {
+			return c
+		}
+	}
+	return nil
+}
+
+func setCondition(mabc *v1.MultiArchBuildConfig, cond *metav1.Condition) {
+	for i := range mabc.Status.Conditions {
+		current := &mabc.Status.Conditions[i]
+		if current.Type == cond.Type {
+			mabc.Status.Conditions[i] = *current
```

Comment:
This is wrong. You shouldn't mutate existing conditions.

Reply:
I'm not sure it's wrong, see [here](https://github.com/kubernetes/kubernetes/blob/8f45b64c93563ef682f24d1f6300679d03d946f1/pkg/controller/replicaset/replica_set_utils.go#L153-L165) as an example.

Reply:
The example you shared is a completely different case and its happening in the replicaSets for a reason. We don't need to update existing conditions because we need to follow the flow of what actually happened in the mabc.

Reply:
> we need to follow the flow of what actually happened in the mabc
That's fine.

Reply:
Btw, at some point, we need to start looking at how those objects will be maintained in the cluster. I am not sure if applying a new version of the mabc will keep the older conditions

Reply:
We can address this problem in a future PR.